[unix_http_server]
file=/tmp/supervisor_nexusai.sock

[supervisord]
pidfile=/tmp/supervisord_nexusai.pid
nodaemon=false

[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[supervisorctl]
serverurl=unix:///tmp/supervisor_nexusai.sock

[program:migrations]
command=python scripts/migrations.py
autostart=true
autorestart=false
startretries=3
stdout_logfile=logs/supervisor/migrations.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/migrations_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:import_documents_to_vdb]
command=python scripts/import_documents_to_vdb.py
autostart=true
autorestart=false
startretries=3
stdout_logfile=logs/supervisor/import_documents_to_vdb.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/import_documents_to_vdb_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:api]
command=python app.py
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/api.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/api_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:websocket]
command=python websocket.py
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/websocket.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/websocket_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:meeting]
command=python task/chatroom_run.py
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/meeting.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/meeting_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:workflow]
command=python task/workflow_run.py
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/workflow.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/workflow_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:celery]
command=python celery_app.py
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/celery.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/celery_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5

[program:web]
command=/bin/bash -c "cd web && npm run start"
autostart=true
autorestart=true
startretries=3
stdout_logfile=logs/supervisor/web.log
stdout_logfile_maxbytes=10MB
stdout_logfile_backups=5
stderr_logfile=logs/supervisor/web_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=5